File: ScriptGraph/TemplateScriptNode/Log/CGLog.js
/**
 * @file CGLog.js
 * @author xuyuan
 * @date 2021/10/15
 * @brief CGLog.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const Amaz = effect.Amaz;
const {BaseNode} = require('./BaseNode');

class CGLog extends BaseNode {
  constructor() {
    super();
  }

  execute(index) {
    if (this.inputs[1]) {
      const logText = this.inputs[1]();
      if (logText) {
        console.log('CREATOR_GRAPH_TAG: ', log);
      }
    }
    if (this.nexts[0]) {
      this.nexts[0]();
    }
  }
}

exports.CGLog = CGLog;

File: ScriptGraph/TemplateScriptNode/Log/CGLog.lua
local CGLog = CGLog or {}
CGLog.__index = CGLog

function CGLog.new()
    local self = setmetatable({}, CGLog)
    self.nexts = {}
    self.inputs = {}
    return self
end

function CGLog:setInput(index, func)
    self.inputs[index] = func
end

function CGLog:setNext(index, func)
    self.nexts[index] = func
end

function CGLog:execute(index)
    if self.inputs[1] then
        local log = self.inputs[1]()
        if log ~= nil then
            Amaz.LOGE("CREATOR_GRAPH_TAG", log)
        end
    end
    if self.nexts[0] then
        self.nexts[0]()
    end
end

return CGLog
